build/vsts_build.yaml

```diff
@@ -183,6 +183,20 @@ jobs:
       optimizationInputsDropNamePrefix: OptimizationInputs/$(System.TeamProject)/$(Build.Repository.Name)
     condition: "and(succeeded(), eq(variables['BuildRTM'], 'false'), eq(variables['IsOfficialBuild'], 'true'))"
 
+  - task: PowerShell@1
+    displayName: "Install dotnet-format"
+    inputs:
+      scriptType: "inlineScript"
+      inlineScript: |
+        (dotnet tool install --tool-path $(System.DefaultWorkingDirectory) dotnet-format) -or (dotnet tool update --tool-path $(System.DefaultWorkingDirectory) dotnet-format)
```

Comment:
I think we can use global(local tools) and enable it locally as well. 
That seems like the right approach to me.

Reply:
@nkolev92 can you explain what you mean? I'm not sure what you mean specifically by gloda/local/enabling locally/etc.

@zivkan the reason it was failing before is because `dotnet tool install` fails if the tool is already installed, but `dotnet tool update` fails if the tool isn't already installed (tested this locally). So this way we have a fallback. Frankly, I'd rather `dotnet tool install` have a `--force` option that installs unconditionally. I also definitely want the `dotnet tool update` there because I don't want us running a super-stale version? Maybe I don't know how dotnet tools added at a solution level should work, though, and we might be able to just take on a dependency on dotnet-format and have it magically install on restore? idkidk. I'll read up on it, ig

Reply:
I just tried locally, and using the built-in local tools was trivial. `dotnet new tool-manifest` to create the tool manifest file. `dotnet tool install dotnet-format` to add `dotnet-format` to the tool manifest. Then `dotnet tool restore` seems to work and $LASTEXITCODE return 0 whether or not the tool was already installed.

I'm still mostly concerned about the dev branch/"official" builds. The more I think about it, the more I think dotnet-format should only run on "private"/PR builds, and not merged commits. Bad formatting is not a good reason to break a merged commit.

Reply:
Thank you! I think that did it.

I **think** I did the only-run-on-PR-builds condition right, too?